pygmt/datasets/earth_relief.py

```diff
@@ -54,8 +83,23 @@ def load_earth_relief(resolution="01d", registration=None):
             "gridline-registered grid is available."
         )
 
-    fname = which(f"@earth_relief_{resolution}{reg}", download="a")
-    grid = xr.open_dataarray(fname)
+    # different ways to load tiled and non-tiled earth relief data
+    if resolution in non_tiled_resolutions:
+        fname = which(f"@earth_relief_{resolution}{reg}", download="a")
+        with xr.open_dataarray(fname) as dataarray:
+            grid = dataarray.load()
+            _ = grid.gmt  # load GMTDataArray accessor information
+    elif resolution in tiled_resolutions:
+        # Titled grid can't be sliced.
+        # See https://github.com/GenericMappingTools/pygmt/issues/524
+        if region is None:
+            raise GMTInvalidInput(
+                "region is required for high resolution (<='05m') Earth relief grid"
+            )
+        grid = grdcut(f"@earth_relief_{resolution}{reg}", region=region)
```

Comment:
Sorry, one more thing. What if a user wants to get a non_tiled_resolution grid **with** a **region**? The current code doesn't seem to handle this case.

Reply:
The initial design was to allow users to specify a subregion for all resolutions. However, due to #524, the returned grids can't support slice operation, which makes three or more tests fail. So I have to do different ways for tiled and non-tiled resolutions, to keep all tests pass.

Reply:
If a user really want it, they need to call `load_earth_relief`, then `grdcut`, or using xarray slice operation.

I'm also OK to add "region" to all resolutions, and declare that "load_earth_relief() grids don't support slice operation" as a bug.

Reply:
Right, that's tricky. But, an unsuspecting user might pass in a `region` argument for resolutions `06m` and above, and it would just be silently ignored (and a full grid is returned)! We should at least raise a warning or `NotImplementedError` for this case.

Reply:
Sorry, slow reply. A workaround might be to just set the `registration and gtype` directly, since we know it from the user input to `load_earth_relief`.

Reply:
> A workaround might be to just set the `registration and gtype` directly after calling `grdcut`, since we know it from the user input to `load_earth_relief`.

Yes, it may make codes a little complicated. If `registration=None`, then the [default earth relief grid registration](https://docs.generic-mapping-tools.org/dev/datasets/remote-data.html#id1) is `pixel` except '03s' and '01s'.

Reply:
If I remember it correctly, I can't just do
```
grid = grdcut("@earth_relief_01d_p", region="0/100/0/30")
grid.gmt.registration = 1
grid.gmt.gtype = 1
```
because `grid.gmt.registration = 1` will run `grdinfo` first, then override the value by `1`. Running `grdinfo` fails because the temporary grid is gone.

Reply:
Actually you can just get it from the `fname`, should either be a `_p` or `_g` at the end. `gtype` would always be Geographic so that's easy. But to be honest, we should probably fix #524 properly.

Reply:
Ah right, then this workaround won't work. So we'll definitely have to fix #524 first.

Reply:
I don't see an easy fix for #524. Perhaps the best compromise is 
> raise a warning or NotImplementedError for this case.

Reply:
Ok, let's compromise on this for now. I don't like the fact that `05m`(and higher resolution) grids which are `grdcut` won't work with `grdimage` or `grdview`, but we can fix this before doing a proper release.

Reply:
Actually, if we remove/pop the tempfile from `grid.encoding["source"]` **after** running `grdcut`, we can then **set** the registration and gtype without getting an error. Do you think it's wise to apply this hack?

Reply:
> that `05m`(and higher resolution) grids which are `grdcut` won't work with `grdimage` or `grdview`, but we can fix this before doing a proper release.

I don't understand it. Why do you say it doesn't work?

Reply:
> I don't understand it. Why do you say it doesn't work?

Sorry, it will work actually, I forgot for a second that we applied the workaround in #539.

Reply:
> Actually, if we remove/pop the temp.nc file from `grid.encoding["source"]` **after** running `grdcut`, we can then **set** the registration and gtype without getting an error. Do you think it's wise to apply this hack?

You mean applying the hack only in `load_earth_relief()` function? Sounds a good idea.

Reply:
Implemented locally, but I'm getting this error:
```python
_________________________________________ [doctest] pygmt.clib.conversion.dataarray_to_matrix _________________________________________
060     >>> matrix.flags.c_contiguous
061     True
062     >>> # Using a slice of the grid, the matrix will be copied to guarantee
063     >>> # that it's C-contiguous in memory. The increment should be unchanged.
064     >>> matrix, region, inc = dataarray_to_matrix(grid[10:41,30:101])
065     >>> matrix.flags.c_contiguous
066     True
067     >>> print(matrix.shape)
068     (31, 71)
069     >>> print(region)
Expected:
    [-150.0, -79.0, -80.0, -49.0]
Got:
    [-149.5, -79.5, -79.5, -49.5]
```

Reply:
> I don't see an easy fix for #524. Perhaps the best compromise is
> 
> > raise a warning or NotImplementedError for this case.

RIght, let's just raise a warning here then, having layers and layers of workarounds isn't good anyway.